cloudfunctions/candidate_gmail_messages/cloudfunction.go

```diff
@@ -243,16 +243,31 @@ func handler(ctx context.Context, e event.Event) error {
 		return handleError("error creating cloud function", err)
 	}
 
-	err = cf.proccessMessages(payload.Messages)
+	err = cf.processMessages(payload.Messages)
 	if err != nil {
 		return handleError("error processing messages", err)
 	}
 
 	return nil
 }
 
-func (cf *CloudFunction) proccessMessages(messageIDs []string) error {
+func (cf *CloudFunction) processMessages(messageIDs []string) error {
+	mlServiceBaseURL := os.Getenv("ML_SERVICE_URL")
+	idTokenSource, err := idtoken.NewTokenSource(cf.ctx, mlServiceBaseURL)
+	if err != nil {
+		return err
+	}
+	idToken, err := idTokenSource.Token()
+	if err != nil {
+		return err
+	}
+	parser := ml.NewService(cf.ctx, ml.NewServiceArg{
+		BaseURL:   mlServiceBaseURL,
+		AuthToken: idToken.AccessToken,
+	})
```

Comment:
You already have access to the ML service via `cf.classifier`...how about renaming it to `cf.model` to make it clear that it can do more than classify?